netty-reactive-streams-http/src/main/java/com/typesafe/netty/http/HttpStreamsServerHandler.java

```diff
@@ -142,7 +142,11 @@ protected void unbufferedWrite(ChannelHandlerContext ctx, HttpStreamsHandler<Htt
                 close = true;
                 continueExpected = false;
             }
-            if (!HttpHeaders.isContentLengthSet(out.message) && !HttpHeaders.isTransferEncodingChunked(out.message)) {
+            // According to RFC 7230 a server MUST NOT send a Content-Length or a Transfer-Encoding when the status
+            // code is 1xx or 204, also a status code 304 may not have a Content-Length or Transfer-Encoding set.
+            int status = out.message.getStatus().code();
+            if (!HttpHeaders.isContentLengthSet(out.message) && !HttpHeaders.isTransferEncodingChunked(out.message)
+                    && !(status < 200 || status == 204 || status == 304)) {
```

Comment:
In fact, to make this more clearer, it'd be good to extract this logic out into another method, eg, `canHaveBody`.